pkg/features/kube_features.go

```diff
@@ -1136,7 +1137,7 @@ var defaultKubernetesFeatureGates = map[featuregate.Feature]featuregate.FeatureS
 
 	PortForwardWebsockets: {Default: false, PreRelease: featuregate.Alpha},
 
-	ProcMountType: {Default: false, PreRelease: featuregate.Alpha},
+	ProcMountType: {Default: false, PreRelease: featuregate.Beta},
```

Comment:
Beta features should default to true, unless there's a very good reason not to.

Reply:
I followed [userns](https://github.com/kubernetes/kubernetes/blob/master/pkg/features/kube_features.go#L1204) . I'm happy to update both but I figure this should be in the same enabled state as that

Reply:
updated to have both enabled by default

Reply:
Can you pull the feature gate promotion out of this PR?

The validation change in isolation looks pretty reasonable, but I need to think about the interaction with the user namespace feature (both in terms of what n-1 API servers will allow through validation once objects exist with data in this field, and in terms of what will happen if the user namespace feature gets disabled, clearing out that field while keeping this procMount field).

procMount validation relying on hostUser field data may mean we have to sequence feature graduation between the two features.

Reply:
removed it! my ideal is they graduate to beta together, though I'd like this to move forward more

> in terms of what will happen if the user namespace feature gets disabled, clearing out that field while keeping this procMount field

WDYT about an option where we clear this field if UserNamespacesSupport and ProcMountType isn't enabled?

Reply:
Thinking briefly about this, I _think_ the following are sufficient (would be good to get a check of that from someone like @deads2k during PRR of the beta promotion):

* All API servers in the cluster have the validation added in this PR in place
* All API servers in the cluster prevent the feature-gate combination `UserNamespacesSupport=false,ProcMountType=true`

We can make the second change in 1.30. We can't go back in time and make 1.29 servers have this validation or enforce that feature-gate pairing.